lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js

```diff
@@ -228,6 +229,39 @@ class UnusedBytes extends Audit {
     };
   }
 
+  /**
+   * Utility function to estimate transfer size and cache calculation.
+   * @param {Map<string, number>} transferRatioByUrl
+   * @param {string} url
+   * @param {LH.Artifacts} artifacts
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {LH.Crdp.Network.ResourceType=} resourceType
+   */
+  static async estimateTransferRatio(transferRatioByUrl, url, artifacts,
+      networkRecords, resourceType) {
+    let transferRatio = transferRatioByUrl.get(url);
+    if (transferRatio !== undefined) return transferRatio;
+
+    const mainDocumentRecord = await NetworkAnalyzer.findMainDocument(networkRecords);
+    const networkRecord = url === artifacts.URL.finalUrl ?
+      mainDocumentRecord :
+      networkRecords.find(n => n.url === url);
+    const script = artifacts.ScriptElements.find(script => script.src === url);
+
+    if (!script || script.content === null) {
+      // Can't find content, so just use 1.
```

Comment:
now that I'm looking at it with fresh eyes this feels wrong, should probably do something like borrowing ratio from the main resource or something that `estimateTransferSize` does